src/extension-test/unit/debugger/util-test.ts

```diff
@@ -17,63 +16,119 @@ function getCoordinates(text: string): (string | number)[] {
     }
 
     it('simple example', () => {
-      expectBreakpointToBeFound('simple.clj');
+      expectBreakpointToBeFound(`;; 3, 2
+      (defn simple [x]
+        (+ 1 #break x|))`);
     });
 
     it('function shorthand', () => {
-      expectBreakpointToBeFound('fn-shorthand.clj');
+      expectBreakpointToBeFound(`;; 3, 1, 2, 2
+      (defn fn-shorthand
+        [s x]
+        (filter #(= % #break x|) s))`);
     });
 
-    it('map', () => {
-      expectBreakpointToBeFound('map.clj');
-    });
+    describe('maps', () => {
+      it('map', () => {
+        expectBreakpointToBeFound(`;; 3, "[1 (+ 1 2)]", 2
+        (defn test-map
+          [x]
+          {:hello "world" [1 (+ 1 2)] (+ 5 #break x|)})`);
+      });
 
-    it('metadata symbol', () => {
-      expectBreakpointToBeFound('metadata-symbol.clj');
-    });
+      it('metadata map', () => {
+        expectBreakpointToBeFound(`;; 3, 2, 2, 2
+        (defn test-metadata-symbol
+          [x]
+          (let [y x]
+            ^{:hello "world"}
+            (+ x ^{:inner "meta"} (+ 1 #break y|))))`);
+      });
 
-    it('metadata map', () => {
-      expectBreakpointToBeFound('metadata-map.clj');
+      it('metadata map last sexp', () => {
+        expectBreakpointToBeFound(`;; 3, 2, 2, 2
+        #dbg
+         (defn test-metadata-symbol
+           [x]
+           (let [y x]
+             ^{:hello "world"}
+             (+ x ^{:inner "meta"} (+ 1 y|))))`);
+      });
     });
 
-    it('metadata map last sexp', () => {
-      expectBreakpointToBeFound('metadata-map-last.clj');
+    it('metadata symbol', () => {
+      expectBreakpointToBeFound(`;; 3, 2, 2
+      (defn test-metadata-symbol
+        [x]
+        (let [y x]
+          (+ ^String x #break ^String y|)))`);
     });
 
     it('ignored forms', () => {
-      expectBreakpointToBeFound('ignored-forms.clj');
+      expectBreakpointToBeFound(`;; 3, 1, 1
+      (defn test-ignore
+        [x]
+        (let [y #_#_#_2 3 4 #break x|]
+          (+ x y)))`);
     });
 
     it('syntax quote', () => {
-      expectBreakpointToBeFound('syntax-quote.clj');
+      expectBreakpointToBeFound(`;; 3, 2, 1, 2, 1, 2
+      (defn test-syntax-quote
+        [y]
+        \`{:hello ~(+ 1 #break y|)})`);
+    });
+
+    describe('derefs', () => {
+      it('atom', () => {
+        expectBreakpointToBeFound(`;; 3, 1
+        #dbg
+        (defn boom
+          []
+          @!a|)`);
+      });
+
+      it('atom then something', () => {
+        expectBreakpointToBeFound(`;; 4, 2, 1
+          #dbg
+          (defn boom
+            []
+            @!a|
+            (let [x 1]
+              (inc x|)))`);
+      });
+
+      // TODO: Figure out why this test fails.
+      // It works interactively.
+      // And the coor given to us from cider-nrepl is really [3, 1]
+      xit('deref call', () => {
+        expectBreakpointToBeFound(`;; 3, 1
+        #dbg
+        (defn boom
+          []
+          @(f)|)`);
+      });
```

Comment:
I don't understand why this test fails. Do you, @jpmonettas?

Reply:
Sorry, no idea, I never wrote Calva tests , not sure how that pipe thing is supposed to work

Reply:
The pipe is placing the document cursor. And these tests start the text with the `coor`. When I run this interactively with the repl message log on I can see that the cursor is exactly there when the debugger shows the ref value and that it has responded with `(3, 1)` coor.

But still a bit unfair to expect you to shed light on this, since it probably is just me having misunderstood something. 😄  Will experiment a bit more with it.

Reply:
If you're curious about the test mechanics, have a look at the rest of the tests in this file and it'll probably clear up a bit.

Reply:
What about the test we were doing before with `@a` instead of `@(f)` ?

Reply:
Ah, thanks. That's a typo. Now fixed. It worked because I think it is interpreted as a selection and the default is that the cursor is at the end of the selection from left to right.